packages/web/components/complex/asset-variants-conversion/asset-variants-conversion-modal.tsx

```diff
@@ -0,0 +1,431 @@
+import { useTranslation, useWindowSize } from "~/hooks";
+import { getSwapMessages, QuoteType } from "~/hooks/use-swap";
+import { ModalBase } from "~/modals";
+import { useStore } from "~/stores";
+import { api } from "~/utils/trpc";
+
+export const useAssetVariantsModalStore = create<{
+  isOpen: boolean;
+  setIsOpen: (value: boolean) => void;
+}>((set) => ({
+  // isOpen: false,
+  isOpen: true,
+  setIsOpen: (value: boolean) => set({ isOpen: value }),
+}));
+
+export const AssetVariantsConversionModal = () => {
+  const { isOpen, setIsOpen } = useAssetVariantsModalStore();
+
+  return (
+    <ModalBase
+      isOpen={isOpen}
+      onRequestClose={() => setIsOpen(false)}
+      title={
+        <h6 className="h6 mt-1 w-full self-center text-center">
+          Standardize Assets
+        </h6>
+      }
+      className="bg-osmoverse-900"
+    >
+      <AssetVariantsConversion onRequestClose={() => setIsOpen(false)} />
+    </ModalBase>
+  );
+};
+
+interface AssetVariantsConversionProps {
+  onRequestClose: () => void;
+}
+
+const AssetVariantsConversion = observer(
+  ({ onRequestClose }: AssetVariantsConversionProps) => {
+    const { accountStore, chainStore } = useStore();
+    const osmosisChainId = chainStore.osmosis.chainId;
+
+    const account = accountStore.getWallet(accountStore.osmosisChainId);
+    const { isMobile } = useWindowSize();
+    const [checkedVariants, setCheckedVariants] = useState<AssetVariant[]>([]);
+    const { t } = useTranslation();
+
+    const {
+      data: allocationData,
+      error: allocationError,
+      isLoading: isAllocationLoading,
+    } = api.local.portfolio.getAllocation.useQuery(
+      {
+        address: account?.address ?? "",
+      },
+      {
+        enabled: !!account?.address,
+        refetchOnWindowFocus: false,
+        onSuccess: (data) => {
+          if (data?.assetVariants) {
+            setCheckedVariants(
+              data?.assetVariants?.map((variant) => variant ?? {}) ?? []
+            );
+          }
+        },
+      }
+    );
+
+    // console.log("Asset Variants:", allocationData?.assetVariants);
+
+    const prepareRouteInputs = useMemo(
+      () =>
+        allocationData?.assetVariants?.map((variant) => ({
+          tokenInDenom: variant.asset?.coinMinimalDenom ?? "",
+          tokenInAmount: variant.amount.toCoin().amount,
+          tokenOutDenom: variant.canonicalAsset?.coinMinimalDenom ?? "",
+        })) ?? [],
+      [allocationData]
+    );
+
+    console.log("prepareRouteInputs", prepareRouteInputs);
+
+    const apiUtils = api.useUtils();
+
+    const quotes = useQueries({
+      queries: prepareRouteInputs.map(
+        ({ tokenInDenom, tokenInAmount, tokenOutDenom }) => ({
+          queryKey: [
+            "routeTokenOutGivenIn",
+            tokenInDenom,
+            tokenInAmount,
+            tokenOutDenom,
+          ],
+          queryFn: () =>
+            apiUtils.local.quoteRouter.routeTokenOutGivenIn.fetch(
```

Comment:
BTW you can can use useQueries with tRPC

https://trpc.io/docs/client/react/useQueries


Like:
```
 api.useQueries((t) =>
  prepareRouteInputs.map(({ tokenInDenom, tokenInAmount, tokenOutDenom }) =>
    t.local.quoteRouter.routeTokenOutGivenIn({
      tokenInDenom,
      tokenInAmount,
      tokenOutDenom,
    })
  )
);
```